src/Elasticsearch/Endpoints/AbstractEndpoint.php

```diff
@@ -60,14 +64,18 @@
     /**
      * @param Transport $transport
      */
-    public function __construct($transport)
+    public function __construct($transport, RequestFactory $requestFactory = null)
     {
         $this->transport = $transport;
+        $this->requestFactory = $requestFactory;
     }
 
     /**
      * @throws \Exception
      * @return array
+     *
+     * @deprecated Performing a request directly on an endpoint is deprecated since 2.2 and will be removed in 3.0,
+     *             you should instead use the createRequest method
      */
     public function performRequest()
```

Comment:
Will continue this and begin to replace the performRequest by createRequest everywhere it is done.

Also the performRequest use an options parameters to set specific options to the underlying http client, this is not possible anymore. 

AFAIK it was only used to for verbose in some parts, but don't know the usage it was done by users ?

> It might be nice to refactor one of the smaller namespaces, like NodesNamespace just to ensure the tests pass and everything works as expected? Then the rest we can do as time allows?

Ok for me

Want also to change the client to have something which respect more PSR7 interface, but this will be another PR, in the meantime i will transform back the PSR7 request into parameters for guzzlering.